### PR TITLE
todo test for GH 18669 (dereferencing ternary skips autovivification)

### PR DESCRIPTION
**Pull request by 1-1sam**
This PR adds a todo test for #18669, which still appears to be broken.

### CODE REVIEW

File: t/run/todo.t [hunk 1]
```diff
@@ -101,6 +101,27 @@ TODO: {
...
+TODO: {
+    local $::TODO = 'GH 18669';
+
+    my $results = fresh_perl(<<~'EOF', {});
+        my $x = { arr => undef };
+        push(@{ $x->{ decide } ? $x->{ not_here } : $x->{ new } }, "mana");
+        print $x->{ new }[0];
+        EOF
+    is($?, 0, "No assertion failure");
+    is($results, 'mana', 'push on non-existant hash entry from ternary autovivifies array ref');
```
Comment (jkeenan) [inline t/run/todo.t]: Spelling: `s/non-existant/non-existent/`

**Review by Imran-imtiaz48 — COMMENTED** [review]
The code in this test script is well-structured for verifying Perl's autovivification behavior with hash and array references. It correctly checks whether pushing to a non-existent hash entry (using a ternary to conditionally assign an array reference) leads to automatic creation of the necessary data structure. The test case with print $x->{ new }[0]; confirms that Perl handles this as expected by producing the value 'mana', and the use of is($?, 0, ...) ensures there’s no runtime error or assertion failure. In the second snippet, the test initializes a hash with an explicit undef value for the arr key, likely preparing for further exploration of edge cases around autovivification. Overall, the code is clean, adheres to testing best practices using Test::More, and provides meaningful checks to validate Perl’s internal behaviors around reference handling. This makes it a valuable contribution to a test suite focused on Perl’s reference semantics.

**Comment by khwilliamson** [comment]
@1-1sam the second commit needs to be squashed into the first one, so that this is a single commit
